packages/ui/react-ui-stack/src/components/Section.tsx

```diff
@@ -53,128 +53,167 @@ export type StackItem = MosaicDataItem &
+                  <Icon className={mx(getSize(5), 'transition-opacity')} />
+                </DropDownMenuDragHandleTrigger>
+                <DropdownMenu.Portal>
+                  <DropdownMenu.Content>
+                    <DropdownMenu.Viewport>
+                      <DropdownMenu.Item onClick={onNavigate} data-testid='section.navigate-to'>
+                        <ArrowSquareOut className={mx(getSize(5), 'mr-2')} />
+                        <span className='grow'>{t('navigate to section label')}</span>
+                      </DropdownMenu.Item>
+                      <DropdownMenu.Item onClick={() => onDelete?.()} data-testid='section.remove'>
+                        <X className={mx(getSize(5), 'mr-2')} />
+                        <span className='grow'>{t('remove section label')}</span>
+                      </DropdownMenu.Item>
+                    </DropdownMenu.Viewport>
+                    <DropdownMenu.Arrow />
+                  </DropdownMenu.Content>
+                </DropdownMenu.Portal>
+              </DropdownMenu.Root>
+              <CollapsiblePrimitive.Trigger asChild>
+                <Button variant='ghost' data-state='' classNames={sectionActionDimensions}>
+                  <span className='sr-only'>{t(collapsed ? 'expand label' : 'collapse label')}</span>
+                  {collapsed ? <CaretUpDown className={getSize(4)} /> : <CaretDownUp className={getSize(4)} />}
                 </Button>
-              </DropdownMenu.Trigger>
-
-              <DropdownMenu.Portal>
-                <DropdownMenu.Content>
-                  <DropdownMenu.Viewport>
-                    <DropdownMenu.Item onClick={onNavigate} data-testid='section.navigate-to'>
-                      <ArrowSquareOut className={mx(getSize(5), 'mr-2')} />
-                      <span className='grow'>{t('navigate to section label')}</span>
-                    </DropdownMenu.Item>
-                    <DropdownMenu.Item onClick={() => onDelete?.()} data-testid='section.remove'>
-                      <X className={mx(getSize(5), 'mr-2')} />
-                      <span className='grow'>{t('remove section label')}</span>
-                    </DropdownMenu.Item>
-                  </DropdownMenu.Viewport>
-                  <DropdownMenu.Arrow />
-                </DropdownMenu.Content>
-              </DropdownMenu.Portal>
-            </DropdownMenu.Root>
+              </CollapsiblePrimitive.Trigger>
+            </div>
+
+            {/* Main content */}
+
+            <ListItem.Heading
+              classNames={
+                collapsed
+                  ? ['grid grid-rows-subgrid grid-cols-subgrid items-center rounded-sm mlb-1 mie-1', focusRing]
+                  : 'sr-only'
+              }
+              {...(collapsed && { ...sectionContentGroup, tabIndex: 0 })}
+            >
+              {/* TODO(thure): This needs to be made extensible; Markdown document titles especially are difficult */}
```

Comment:
what do you think about just using a separate `Surface` here? Rather than trying to recompute what the title should be just defer to a plugin that knows how to render it. Stack could provide this title span as a fallback?

Reply:
I was wondering if this might be the right approach, I can give it a try!

Reply:
another option might be to just find the associated graph node and use it's label...both seems to have tradeoffs to me.

if you use the surface you may end up with some duplicate logic within plugins, but it nicely decouples things and also would allow you to render more than just text as a heading potentially

if you use the graph, you're able to just use existing labels, but it couples stack to knowing about the graph and means that all sections need to be in the graph (however, stack knowing about the graph doesn't seem like the end of the world as far as coupling goes and sections will likely always be in the graph anyways)

Reply:
fwiw everything that gets rendered in the main area is the datum of a graph node, its just that the graph node isn't what is passed to the surface...

Reply:
I'd say just do what makes most sense for now. We will revisit soon because after the schema work lands stacks & folders are going to devolve into collections.

Reply:
Ah, I remember why I didn’t use `Surface`, it’s part of `app-framework` which no other UI package uses. This would set a new precedent, but I think it doesn’t need to and shouldn’t.